core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/DatasetResource.scala

```diff
@@ -561,6 +576,54 @@ object DatasetResource {
   case class DatasetNameModification(did: UInteger, name: String)
 
   case class DatasetDescriptionModification(did: UInteger, description: String)
+
+  private def calculateSize(did: UInteger, versionHash: Option[String] = None): Long = {
+    Try {
```

Comment:
add comments that is versionHash is not given, calculate the latest dataset size